### PR TITLE
[WIP] financial reports, sorrento wrap-up

### PR DESCRIPTION
**Pull request by lepistone**

### CODE REVIEW

**Comment by lepistone** [comment]
@andrea4ever I noticed that the open invoices wizard has references to `until_date` but a field `until_date` is not present. Do you think we should add the field or remove references to it?

Thanks!

File: account_financial_report_qweb/wizard/general_ledger_wizard.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        <group name="moves">
+                            <field name="target_move" widget="radio"/>
+                        </group>
+                    </group>
+                    <label for="account_ids"/>
+                    <field name="account_ids" nolabel="1"/>
+                    <footer>
+                        <button name="button_view" string="View" type="object" default_focus="1" class="oe_highlight"/>
+                        or
+                        <button name="check_report_xlsx" string="Export XLSX" type="object"/>
```
Comment (paulius-sladkevicius) [inline account_financial_report_qweb/wizard/general_ledger_wizard.xml:36]: check_report_xlsx() is defined under ledger.report.wizard.line, not ledger.report.wizard

**Comment by paulius-sladkevicius** [comment]
We know that it's still in WIP, but we tried to test it out already (love them on v8) and want to share errors which we faced:

Printing Partner Balance & Trial Balance:

```
Traceback (most recent call last):
File "/home/odoo/odoo9/odoo/openerp/http.py", line 643, in handle_exception
return super(JsonRequest, self)._handle_exception(exception)
File "/home/odoo/odoo9/odoo/openerp/http.py", line 680, in dispatch
result = self._call_function(**self.params)
File "/home/odoo/odoo9/odoo/openerp/http.py", line 316, in _call_function
return checked_call(self.db, *args, **kwargs)
File "/home/odoo/odoo9/odoo/openerp/service/model.py", line 118, in wrapper
return f(dbname, *args, **kwargs)
File "/home/odoo/odoo9/odoo/openerp/http.py", line 309, in checked_call
result = self.endpoint(*a, **kw)
File "/home/odoo/odoo9/odoo/openerp/http.py", line 959, in call
return self.method(*args, **kw)
File "/home/odoo/odoo9/odoo/openerp/http.py", line 509, in response_wrap
response = f(*args, **kw)
File "/home/odoo/odoo9/enterprise/web/controllers/main.py", line 911, in call_button
action = self._call_kw(model, method, args, {})
File "/home/odoo/odoo9/enterprise/web/controllers/main.py", line 899, in _call_kw
return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
File "/home/odoo/odoo9/odoo/openerp/api.py", line 250, in wrapper
return old_api(self, *args, **kwargs)
File "/home/odoo/odoo9/odoo/openerp/api.py", line 381, in old_api
result = method(recs, *args, **kwargs)
File "/home/odoo/odoo9/odoo/addons/account/wizard/account_report_common.py", line 39, in check_report
return self._print_report(data)
File "/home/odoo/odoo9/odoo/addons/account/wizard/account_report_common.py", line 28, in _print_report
raise (('Error!'), _('Not implemented.'))
```

Other clear issue commented directly under code.

**Comment by lepistone** [comment]
Thanks for the review @komsas . This indeed needs some love, and there are still some general issues with this approach we haven't figured out yet. I should have some time for this this or next week.

File: account_financial_report_qweb/wizard/aged_partner_balance_wizard.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Author: Damien Crier, Andrea Stirpe, Kevin Graveman, Dennis Sluijk
+# Copyright 2016 Camptocamp SA, Onestein B.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from datetime import datetime
+from openerp.exceptions import Warning as UserError
```
Comment (atchuthan) [inline account_financial_report_qweb/wizard/aged_partner_balance_wizard.py:7]: For version 9.0, it is preferred to use UserError in the imports

File: account_financial_report_qweb/wizard/aged_partner_balance_wizard_view.xml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Comment (atchuthan) [inline account_financial_report_qweb/wizard/aged_partner_balance_wizard_view.xml:2]: openerp and data tags are replaced by odoo tags in 9.0

File: account_financial_report_qweb/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Author: Damien Crier
+# Copyright 2016 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': 'QWeb Financial Reports',
+    'version': '9.0.0.1.0',
```
Comment (atchuthan) [inline account_financial_report_qweb/__openerp__.py:7]: '9.0.0.1.0' -> '9.0.1.0.0'
As it is a major improvement

File: account_financial_report_qweb/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        'wizard/balance_common_wizard_view.xml',
+        'wizard/partner_balance_wizard_view.xml',
+        'wizard/trial_balance_wizard_view.xml',
+        'wizard/balance_sheet_wizard_view.xml',
+        'menuitems.xml',
+        'reports.xml',
+        'report/templates/general_ledger.xml',
+        'report/templates/open_invoice_report.xml'
+    ],
+    'test': [
```
Comment (atchuthan) [inline account_financial_report_qweb/__openerp__.py:36]: could remove unused/empty values

File: account_financial_report_qweb/models/account.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).-
+from openerp import models, fields
+
+
+class AccountAccount(models.Model):
+    _inherit = 'account.account'
+
+    centralized = fields.Boolean(
+        'Centralized',
+        default=False,
```
Comment (atchuthan) [inline account_financial_report_qweb/models/account.py:12]: For a boolean field, it is by default False.  So, no need to explicitly define it.

**Comment by pedrobaeza** [comment]
@atchuthan, I think you're reviewing the wrong PR: https://github.com/OCA/account-financial-reporting/pull/211.

I'm closing this one.

File: account_financial_report_qweb/wizard/aged_partner_balance_wizard.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    @api.onchange('at_date')
+    def onchange_atdate(self):
+        self.until_date = self.at_date
+
+    @api.onchange('until_date')
+    def onchange_untildate(self):
+        # ---- until_date must be always >= of at_date
+        if self.until_date:
+            if self.until_date < self.at_date:
+                raise UserError(
```
Comment (atchuthan) [inline account_financial_report_qweb/wizard/aged_partner_balance_wizard.py:61]: use translation on raise with _(...)

**Comment by atchuthan** [comment]
thanks @pedrobaeza

**Pull request closed without merging**